plugins/parsers/json/parser.go

```diff
@@ -209,6 +212,13 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	}
 
 	var data interface{}
+	if p.escape {
+		s, err := strconv.Unquote(string(buf))
+		if err != nil {
+			return nil, err
```

Comment:
You probably want to add some information to the error to find this location in code when the error occurs.
```suggestion
			return nil, fmt.Errorf("unquoting buffer failed: %v", err)
```